compiler/rustc_errors/src/emitter.rs

```diff
@@ -317,13 +317,12 @@ pub trait Emitter {
         self.render_multispans_macro_backtrace(span, children, backtrace);
 
         if !backtrace {
-            if let Some(macro_kind) = has_macro_spans {
+            if let Some((macro_kind, name)) = has_macro_spans {
+                let descr = macro_kind.descr();
+
                 let msg = format!(
-                    "this {} originates in {} {} \
+                    "this {level} originates in the {descr} `{name}` \
```

Comment:
```suggestion
                    "this {level} originates in {descr} `{name}` \
```

Reply:
I think the current output looks better - e.g. 'this error originates in the attribute macro `test`' vs 'this error originates in attribute macro `test`'

Reply:
I'm not bothered either way :)

It seems like you have to rebless, but beyond that, r=me.